### PR TITLE
fix: consistent mangling for private imports

### LINKED ISSUE #9151: import of internal modules is not working any more >0.21.1
### Describe the bug

With version: marimo 23.1 there is an error when import internal modules (startswith "_").
Works with marimo 21.

I'm importing a local module like that:
```python
from cama import _camboardscan, _camera, _scanner
```
I get a name error: 
``NameError: name _cell_ffec47fb-6ddf-4e37-a31b-0d21dc8967db_camera is not defined```

Workaround would be using aliases but this feels a bit clumsy. As I'm using marimo to explore things and later integrate it maybe into my code I would have to rename everything.

Background: I use on start of development mostly internal modules and providing later on an public interface which makes it easier to mess around with my code without breaking any "public" imports for other users.

### Will you submit a PR?

- [ ] Yes

### Environment

<details>

```
NameError: name '_cell_ffec47fb-6ddf-4e37-a31b-0d21dc8967db_camera' is not defined
```

</details>

### Code to reproduce

_No response_

### PR DESCRIPTION
**Pull request by dmadisetti**
## 📝 Summary

A scoping change introduced in #8762 changed `_` prefixed imports by turning then into privates.
While this is consistent with the remainder of marimo's private pattern, it is technically a breaking change as of 0.22.0, and the edge case was not correctly handled. The change lead to a bug where imports referenced in scope were not correctly addressed. This PR handles the mangling behavior for these import cases.

An alternative PR, for retaining the behavior, is here: https://github.com/marimo-team/marimo/pull/9308 but this requires ref tracking a little differently such that the `_` pattern is properly placed into globals.

closes #9151

### CODE REVIEW

**Review by mscolnick — COMMENTED** [review]
can we also test renaming the private import?

File: marimo/_ast/visitor.py [hunk 1]
```diff
@@ -298,7 +295,13 @@ def _get_alias_name(
                     f"{line} SyntaxError: Importing symbols with `import *` "
                     "is not allowed in marimo."
                 )
-            return basename
+            # Previously we did not mangle. So this is technically breaking
+            # from 0.22.0 (#8762) but now consistent with private naming
+            # conventions.
+            mangled = self._if_local_then_mangle(basename)
```
Comment (mscolnick) [inline marimo/_ast/visitor.py:306]: why did this require actual code changes? I thought it was just to add a verification test?

Comment (dmadisetti) [reply marimo/_ast/visitor.py:306]: The current implementation is broken, The test I put in caught the edge case where a scoped import no longer works.

Comment (mscolnick) [reply marimo/_ast/visitor.py:306]: got it, maybe worth thinking through some other edge cases then (lazy imports etc)

File: marimo/_ast/visitor.py [hunk 2]
```diff
@@ -298,7 +295,13 @@ def _get_alias_name(
...
+            if mangled != basename:
+                node.asname = mangled
```
Comment (akshayka) [inline marimo/_ast/visitor.py:329]: I don't understand this branch. Isn't `node.asname` `None`? Why are we assigning the mangled basename to it? Should it be `name.name = mangled`?

**Review by akshayka — COMMENTED** [review]
Can you make the PR description more specific?

> and the edge case was not correctly handled.

Which edge case?

> The change lead to a bug where imports referenced in scope were not correctly addressed. This PR handles the mangling behavior for these import cases.

What do you mean by not correctly addressed? How does this PR fix it?

File: marimo/_ast/visitor.py (hunk 2 above)
Comment (dmadisetti) [reply marimo/_ast/visitor.py:329]: We can't rename a module but:

`import _private as _mangled_private` is fine

I do think I'm missing the

`import _private.submodule`

case though. Will revise and add a comment.

File: marimo/_ast/visitor.py [hunk 3]
```diff
@@ -298,7 +295,13 @@ def _get_alias_name(
                     f"{line} SyntaxError: Importing symbols with `import *` "
                     "is not allowed in marimo."
                 )
-            return basename
+            # Previously we did not mangle. So this is technically breaking
+            # from 0.22.0 (#8762) but now consistent with private naming
+            # conventions.
+            mangled = self._if_local_then_mangle(basename)
+            if mangled != basename:
```
[the remark this answers is not in the document]
Comment (dmadisetti) [reply marimo/_ast/visitor.py:307]: Thanks

[the remark this answers is not in the document]
Comment (dmadisetti) [reply marimo/_ast/visitor.py:307]: but not the right fix

**Comment by akshayka** [comment]
> Can you make the PR description more specific?
> 
> > and the edge case was not correctly handled.
> 
> Which edge case?
> 
> > The change lead to a bug where imports referenced in scope were not correctly addressed. This PR handles the mangling behavior for these import cases.
> 
> What do you mean by not correctly addressed? How does this PR fix it?

@dylan can you address these comments? This PR is subtle enough that I would like the intended behavior to be very clear.

**Comment by dmadisetti** [comment]
Closing out for simpler method
